src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerWorkspaceFactory.cs

```diff
@@ -34,7 +35,9 @@ public LanguageServerWorkspaceFactory(
 
         var workspace = new LanguageServerWorkspace(hostServicesProvider.HostServices);
         Workspace = workspace;
-        ProjectSystemProjectFactory = new ProjectSystemProjectFactory(Workspace, fileChangeWatcher, static (_, _) => Task.CompletedTask, _ => { });
+        ProjectSystemProjectFactory = new ProjectSystemProjectFactory(
+            Workspace, fileChangeWatcher, static (_, _) => Task.CompletedTask, _ => { },
+            CancellationToken.None); // TODO: do we need to introduce a shutdown cancellation token for this?
```

Comment:
this is what we do elsewhere in the LSP server.  note: it would make sense for the LSP serer to have a centralized service to get a sincel CT representing its lifetime (if servers have a well defined lifetime where they are asked to stop).

Reply:
this should be fine - the lifetime of the project system factory is the process lifetime.  we could get a cancellation token here but not necessary